app/javascript/shared/components/focustrap.jsx

```diff
@@ -0,0 +1,30 @@
+import PropTypes from 'prop-types';
+import { h, Fragment } from 'preact';
+import { useEffect, useRef } from 'preact/hooks';
+import { createFocusTrap } from 'focus-trap';
+import { defaultChildrenPropTypes } from '../../common-prop-types';
+
+const FocusTrap = ({ selector, children }) => {
```

Comment:
I created this wrapper component that uses library 'focus-trap' to trap focus. Ideally I love to  convert it to a hook rather than a component but since it is used by class components too it can't be a hook.